prusti-encoder/src/encoders/impure/fn_wand.rs

```diff
@@ -96,55 +83,67 @@ impl<'vir, E: TaskEncoder> ImpureEncVisitor<'vir, '_, E> {
     }
 }
 
-#[derive(Clone, Debug)]
+type EncodedPledges<'vir> = Vec<EncodedPledge<'vir>>;
+
+#[derive(Clone, Debug, Default)]
 pub struct WandEncOutput<'vir> {
-    context: GParams<'vir>,
-    edges: WandEncEdges,
-    pub generic_to_param: FxHashMap<IndirectKey, Vec<(mir::Local, ty::Ty<'vir>)>>,
-    #[allow(dead_code)]
-    pub pledges: Pledges<'vir>,
-    wands: Vec<(Vec<IndirectKey>, Vec<IndirectKey>, Pledges<'vir>)>,
-}
+    /// Information about the corresponding function. This is wrapped in an
+    /// `Option` in order to have `WandEncOutput: Default`; when explicitly
+    /// constructed, it should always be present.
+    function_data: Option<FunctionData<'vir>>,
```

Comment:
Do we actually need the `Debug` implementation? I would prefer a constructor to having this be an option.

Reply:
nope, removed the option